sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs

```diff
@@ -67,3 +70,18 @@ public UnexpectedDiagnosticException(Location location, string message) : base(m
             Path.GetFullPath(Path.Combine(@"..\..\..\TestCases", path));
     }
 }
+
+#else // Resources cannot be accessed in the same way on .Net Core
+
+namespace SonarAnalyzer.UnitTest.TestFramework
+{
+    public class UnexpectedDiagnosticException : Exception
```

Comment:
maybe document somewhere that unit test debugging differs on netcore vs netframework

Reply:
I don't understand what we gain from this.

Reply:
avoid confusion. For example, if later on, someone will try and run the UTs from a Linux box and would expect a useful debug message, we can tell that person that the Netcore debugging is not as refined as the Net Framework debugging

Reply:
I've realized that we can hardcode the template from English resources for .Net 😄 It could be good enough for now. What do you think?

Reply:
what does "template from English resources for .Net Core" mean?

Reply:
`UnexpectedDiagnosticException` extends the logged information by adding a line which contains the path to the file with errors and the line on which the error occurs. In order to generate this lines it takes two strings from internal resources:
- the word **"at"** by using `mscorlibResources.GetString("Word_At")`
- and the first part of the stack trace, the line which contains the file name and the line number. This is done by `stackTraceFormat = mscorlibResources.GetString("StackTrace_InFileLineNumber")`

The result is a message formatted like: 

> at Test Case in D:\git\sonar-dotnet\sonaranalyzer-dotnet\tests\SonarAnalyzer.UnitTest\TestCases\BooleanCheckInverted.cs:line 22

the same can be achieved by hardcoding the format and the word **"at"**  with the following `$"at Test Case in {path}:line {line}";`.

Since I don't know how we can get the resources on .Net Core, we have to use English words and this means the line will not be translated (to French as example) when running on environments with another cultures.

Reply:
Aaaa yes, I believe it's fine having English messages, it's mainly for our own consumption